src/MongoDB.Driver/Encryption/MongocryptdFactory.cs

```diff
@@ -115,13 +117,7 @@ private bool ShouldMongocryptdBeSpawned(out string path, out string args)
                 }
                 else
                 {
-                    path = string.Empty; // look at the PATH env variable
-                }
-
-                if (!Path.HasExtension(path))
```

Comment:
this is a BC. it's related to the fact that linux files actually can not have any extensions. So we cannot determine whether the last path node is a file or folder anymore. It's possible to fix it via `if Windows/NonWindows` conditions, but maybe it's better the same behavior even though it's a minor BC (which can appear only if a user uses `mongocryptdSpawnPath`)

Reply:
I will check it

Reply:
yep, It looks like it makes sense. But I think Directory.Exists is a bit more suitable here. Done